opentelemetry-sdk/src/opentelemetry/sdk/_metrics/export/__init__.py

```diff
@@ -96,6 +96,36 @@ def shutdown(self) -> None:
         pass
 
 
+class InMemoryMetricReader(MetricReader):
+    """Implementation of :class:`MetricReader` that returns its metrics from :func:`metrics`.
+
+    This is useful for e.g. unit tests.
+    """
+
+    def __init__(
+        self,
+        preferred_temporality: AggregationTemporality = AggregationTemporality.CUMULATIVE,
+    ) -> None:
+        super().__init__(preferred_temporality=preferred_temporality)
+        self._lock = RLock()
+        self._metrics: List[Metric] = []
+
+    def get_metrics(self) -> List[Metric]:
+        """Reads and returns current metrics from the SDK"""
+        with self._lock:
+            self.collect()
+            metrics = self._metrics
+            self._metrics = []
+        return metrics
```

Comment:
Not sure if this method should reset `self._metrics`. I imagine someone may want to just want to know which are the current metrics of the reader, being able to do that repeatedly and expecting them not to change until `collect` is being called again.

Reply:
The idea here is this is a pull exporter with `get_metrics()` being the way to scrape. This sounds like a push exporter version which we could add. Do you have any example use cases in mind for this?

Reply:
Just the use case above, where the user wants to query repeatedly the value of current metrics, I'm ok with adding it later.